dali/kernels/common/split_shape.h

```diff
@@ -59,6 +63,52 @@ void split_shape(SplitFactor& split_factor, const Shape& in_shape, int min_nbloc
   }
 }
 
+/**
+ * @brief returns the dimension index with a split factor > 1
+ */
+template <typename SplitFactor>
+int LastSplitDim(const SplitFactor& split_factor) {
+  int last_split_dim = -1;
+  int ndim = dali::size(split_factor);
+  for (int d = ndim - 1; d >= 0; d--) {
+    if (split_factor[d] > 1) {
+      last_split_dim = d;
+      break;
+    }
+  }
+  return last_split_dim;
+}
+
+/**
+ * @brief Schedule work on a per-block basis
+ */
+template <int Dims, typename SplitFactor, typename ScheduleBlockFunc>
+void ScheduleBlocks(TensorShape<Dims> start, TensorShape<Dims> end,
+                    const SplitFactor& split_factor, int d, int max_split_dim,
+                    ScheduleBlockFunc&& func) {
+  if (d > max_split_dim || d == Dims) {
```

Comment:
```suggestion
  if (d > max_split_dim || d == start.size()) {
```
This will handle DynamicDimensions